code/site/controllers/http.php

```diff
@@ -40,22 +40,35 @@ public function display($cachable = false, $urlparams = array())
 		$name = $app->input->get('app', '', 'CMD');
 
 		$params = JComponentHelper::getParams('com_api');
-
-		// Set CORS header add comma seperated values directaly in input box
-		$cors_urls = $params->get('cors', "*");
-
-		$callMethod = $app->input->server->get('REQUEST_METHOD', '', 'STRING');
+		$callMethod = $app->input->getMethod();
 
 		// Special method for OPTIONS method
 		if ((!empty($params->get("allow_cors"))))
 		{
-			header("Access-Control-Allow-Origin: " . $cors_urls);
+			$corsUrls = $params->get('cors', "*");
+
+			if ($corsUrls === "*")
+			{
+				header("Access-Control-Allow-Origin: " . '*');
+			}
+			else
+			{
+				$httpOrigin = $app->input->server->getString('HTTP_REFERER', '');
```

Comment:
The referer may give a full URL like http://truck.com/jobs and not just http://truck.com 

Is this being taken care of ?